featuretools_tsfresh_primitives/primitives/fft_coefficient.py

```diff
@@ -40,3 +45,115 @@ def function(x):
+            Defaults to 0, which will include the target instance in the window.
+        min_periods (int, optional): Minimum number of observations required for performing calculations
+            over the window. Can only be as large as window_length when window_length is an integer.
+            When window_length is an offset alias string, this limitation does not exist, but care should be taken
+            to not choose a min_periods that will always be larger than the number of observations in a window.
+            Defaults to 1.
+
+    Note:
+        Only offset aliases with fixed frequencies can be used when defining gap and window_length.
+        This means that aliases such as `M` or `W` cannot be used, as they can indicate different
+        numbers of days. ('M', because different months are different numbers of days;
+        'W' because week will indicate a certain day of the week, like W-Wed, so that will
+        indicate a different number of days depending on the anchoring date.)
+
+    Note:
+        When using an offset alias to define `gap`, an offset alias must also be used to define `window_length`.
+        This limitation does not exist when using an offset alias to define `window_length`. In fact,
+        if the data has a uniform sampling frequency, it is preferable to use a numeric `gap` as it is more
+        efficient.
+
+    Docstring source:
+    https://tsfresh.readthedocs.io/en/latest/api/tsfresh.feature_extraction.html#tsfresh.feature_extraction.feature_calculators.fft_aggregated
+
+    Examples:
+        >>> import pandas as pd
+        >>> datetimes = pd.date_range(start="2019-01-01", freq="1D", periods=10)
+        >>> stfftc = ShortTermFftCoefficient(1, "real")
+        >>> stfftc(datetimes, [1, 2, 4, 8, 16, 24, 48, 96, 192, 384]).tolist()
+        [nan, -1.0, -2.0, -4.0, -8.0, -12.0, -20.0, -48.0, -96.0, -192.0]
+
+        We can also control the gap before the rolling calculation.
+
+        >>> stfftc = ShortTermFftCoefficient(1, "real", gap=1)
+        >>> stfftc(datetimes, [1, 2, 4, 8, 16, 24, 48, 96, 192, 384]).tolist()
+        [nan, nan, nan, -2.0, -4.0, -8.0, -12.0, -20.0, -48.0, -96.0]
+
+        We can also control the attribute type.
+
+        >>> stfftc = ShortTermFftCoefficient(1, "imag")
+        >>> stfftc(datetimes, [1, 2, 4, 8, 16, 24, 48, 96, 192, 384]).tolist()
+        [nan, 0.0, 1.7320508075688772, 3.4641016151377544, 6.928203230275509, 6.928203230275509, 20.784609690826528, 41.569219381653056, 83.13843876330611, 166.27687752661222]
+
+    """
+
+    name = "short_term_fft_coefficient"
+    input_types = [
+        ColumnSchema(logical_type=Datetime, semantic_tags={"time_index"}),
+        ColumnSchema(semantic_tags={"numeric"}),
+    ]
+    return_type = ColumnSchema(logical_type=Double, semantic_tags={"numeric"})
+
+    def __init__(self, coeff, attr, window_length=3, gap=0, min_periods=0):
```

Comment:
Similar to my other comment - should we have default values for `coeff` and `attr` here?